base/utils/R/convert.input.R

```diff
@@ -115,17 +115,16 @@ convert.input <- function(input.id, outfolder, formatname, mimetype, site.id, st
       # regular expression specific special characters and site-specific identification is already present through
       # site.id, a regex placeholder is used instead.
       
-      filename_pattern = pattern #pattern is always the name of the meteorological data product (met).
-      if (!is.null(input.args$sitename) || !is.null(site.id)) {
-        filename_pattern = paste(filename_pattern, "[^.]*", sep="\\.") #double backslash for regex
-      }
+      #pattern is always the name of the meteorological data product (met).
+      filename_pattern = paste0(pattern, "\\.([^.]*\\.)?") #double backslash for regex
```

Comment:
It's a placeholder for the site name, which occurs in NOAA GEFS raw data names.  The `?` makes it optional and the  `if (!is.null(input.args$sitename) || !is.null(site.id)) {` unnecessary.  We don't actually care what the site name is, because site names are selected for with a different argument to `dbfile.input.check`; the entire point of the regex argument is to identify the ensemble (#2033).  Using this instead of the if statement makes the regex more flexible in identifying the ensemble.